files.go

```diff
@@ -0,0 +1,108 @@
+package pack
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/pkg/errors"
+
+	"github.com/buildpack/pack/builder"
+	"github.com/buildpack/pack/internal/paths"
+)
+
+// ReadBuilderConfigFile reads a builder configuration from the file path provided
+func ReadBuilderConfigFile(path string) (builder.Config, error) {
+	builderDir, err := filepath.Abs(filepath.Dir(path))
+	if err != nil {
+		return builder.Config{}, err
+	}
+
+	builderConfig := builder.Config{}
+	if _, err = toml.DecodeFile(path, &builderConfig); err != nil {
+		return builderConfig, fmt.Errorf(`failed to decode builder config from file %s: %s`, path, err)
+	}
+
+	for i, bp := range builderConfig.Buildpacks {
+		uri, err := transformRelativePath(bp.URI, builderDir)
+		if err != nil {
+			return builder.Config{}, errors.Wrap(err, "transforming buildpack URI")
+		}
+		builderConfig.Buildpacks[i].URI = uri
+	}
+
+	if builderConfig.Lifecycle.URI != "" {
+		uri, err := transformRelativePath(builderConfig.Lifecycle.URI, builderDir)
+		if err != nil {
+			return builder.Config{}, errors.Wrap(err, "transforming lifecycle URI")
+		}
+		builderConfig.Lifecycle.URI = uri
+	}
+
+	return builderConfig, nil
+}
+
+func transformRelativePath(uri, relativeTo string) (string, error) {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return "", err
+	}
+
+	if parsed.Scheme == "" {
+		if !filepath.IsAbs(parsed.Path) {
+			absPath := filepath.Join(relativeTo, parsed.Path)
+			return paths.FilePathToUri(absPath)
+		}
+	}
+
+	return uri, nil
+}
+
+// ReadEnvFile reads environment variables from the file path provided. The file should have one variable per line,
+// where each variable is of the form 'VAR=VALUE' or 'VAR'. When using the latter value-less form, value will be taken
+// from the current environment. Additional variables of the same form may be provided (and will have precedence over
+// those specified in the file).
+func ReadEnvFile(path string, additionalVars ...string) (map[string]string, error) {
```

Comment:
I know I said I wasn't going to add the additional vars to this method, but I ran into issues due to splitting the associated methods between the `commands` and `pack` package, so I move them all over. It seemed like the simplest way to avoid duplicating the `addEnvVar` functionality.

Reply:
I still feel like this interface is confusing with the additionalVars are a parameter. I feel like something called ReadEnvFile should just concern itself with the file. Why can't the command package have a private helper function that parses the env flags and adds the vars to the map?

Reply:
@ekcasey A private helper is not a problem. It would just have to be duplicated. pack can't import the commands package, and commands cant use it if it lives privately in pack. I could create a third package to decouple, but feels a little tedious.

Reply:
Moved to internal package and added tests